Site Creation: Fix renderer crash for long site names when creating new site

gavande1 opened the pull request:
## Related issues

- Fixes STU-1182

## Proposed Changes

- Added error handling in `handleSiteNameChange` to catch ENAMETOOLONG errors and display a user-friendly error message when site name is too long
- Added tests to verify ENAMETOOLONG error handling works correctly
- The renderer now gracefully handles long site names instead of crashing

## Testing Instructions

1. Open the "Add site" modal
2. Enter a very long site name (e.g., a string of 300+ characters)
3. Verify that instead of crashing, a user-friendly error message is displayed: "The site name is too long. Please choose a shorter site name."
4. Run the test suite: `npm test -- src/hooks/tests/use-add-site.test.tsx`
5. Verify all tests pass, including the new ENAMETOOLONG error handling tests

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

gavande1 committed:
Fix renderer crash for long site names when creating new site

- Add error handling in handleSiteNameChange to catch ENAMETOOLONG errors
- Display user-friendly error message when site name is too long
- Add tests to verify ENAMETOOLONG error handling

Fixes STU-1182

nightnei reviewed (approved):
LGTM and works as expected 👍
<img width="443" height="438" alt="Screenshot 2025-12-30 at 13 29 40" src="https://github.com/user-attachments/assets/209cbbf3-6d31-44a7-b981-8861b4e963c3" />

wojtekn commented on src/hooks/use-add-site.ts:
```diff
@@ -244,30 +244,39 @@ export function useAddSite( options: UseAddSiteOptions = {} ) {
-					)
-				);
-				return;
+			try {
```
As the exception can be thrown by `generateProposedSitePath`, it would be clearer to wrap only this part with `try/catch`.

However, in `generateProposedSitePath`, we already catch the underlying exception and set returned props accordingly, so we could consider moving exception handling there.

gavande1 replied:
Initially, My first approach was to handle exceptions in `generateProposedSitePath`. However, the main issue was with the error message. If we handle the exception in IPC handlers, it prefixes the following error message. 

```
Error invoking remote method 'generateProposedSitePath'
```

if we want to remove this prefix, I would need to add an additional parameter, such as `isError`, to indicate that there was a problem. However, I believe this is unnecessary.

<img width="872" height="522" alt="CleanShot 2025-12-31 at 14 07 53@2x" src="https://github.com/user-attachments/assets/604ea991-823f-404b-b2eb-59f11c8edbe7" />

wojtekn replied:
What if we handled the exception there, returned `isNameTooLong` prop, and set the error in the renderer process? It would be consistent with current error handling.

gavande1 replied:
Yes. To make it consistent we can do that. I will update the PR.

gavande1 replied:
Updated in [34e6d43](https://github.com/Automattic/studio/pull/2336/commits/34e6d431630d283f2fed6b057769b9c58c6650b7)!

gavande1 committed:
Address code review: Move ENAMETOOLONG handling to IPC handler

- Move ENAMETOOLONG error handling to generateProposedSitePath IPC handler
- Simplify hook to display error message from IPC handler
- Update tests to reflect the change

Addresses feedback from code review

gavande1 committed:
Revert "Address code review: Move ENAMETOOLONG handling to IPC handler"

This reverts commit 6931b1fb807d9e414814c81413c9230d78b4a1e2.

gavande1 committed:
Handle ENAMETOOLONG via isNameTooLong prop

katinthehatsite reviewed (approved):
This looks good to me 👍 

I got a visual indication when the site name was too long:

<img width="691" height="464" alt="Image" src="https://github.com/user-attachments/assets/2758a644-28ed-4119-a118-10d550e420d1" />
